crates/sui/src/client_commands.rs

```diff
@@ -1105,8 +1109,18 @@ impl Display for SuiClientCommandResult {
                     writeln!(writer, "{}", parsed_resp)?;
                 }
             }
-            SuiClientCommandResult::Object(object_read) => {
-                let object = unwrap_err_to_string(|| Ok(object_read.object()?));
+            SuiClientCommandResult::Object(object_read, bcs) => {
+                let object = if *bcs {
+                    match object_read.object() {
+                        Ok(v) => {
+                            let bcs_version = bcs::to_bytes(v).unwrap();
```

Comment:
```suggestion
                            let bcs_bytes = bcs::to_bytes(v).unwrap();
```
nit: `bcs_version` might be a bit confusing